Set starter team userLimit to 3 and enforce member limits in api

knolleary opened the pull request:
Closes #781 

This PR modifies the default `starter` team type to have a limit of 3 members.

It also adds additional checks to both the Team and Invitation controllers to ensure the userLimit is honoured.

hardillb reviewed (commented):
Code changes look OK, but need to work out why the tests are failing

knolleary commented:
Fixed the test. This one test resulted in a team having 4 members - which is no longer possible without a proper license. Changed the test to use a different team so as not to hit the limit.
